modules/github-bots/sdk/bot.go

```diff
@@ -0,0 +1,83 @@
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"runtime/debug"
+
+	"github.com/chainguard-dev/clog"
+	"github.com/chainguard-dev/clog/gcp"
+	"github.com/chainguard-dev/terraform-infra-common/modules/github-events/schemas"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/go-github/v61/github"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type Bot interface {
+	Name() string
+}
+
+func Serve(b Bot) {
+	var env struct {
+		Port int `envconfig:"PORT" default:"8080" required:"true"`
+	}
+	if err := envconfig.Process("", &env); err != nil {
+		clog.Fatalf("failed to process env var: %s", err)
+	}
+
+	slog.SetDefault(slog.New(gcp.NewHandler(slog.LevelInfo)))
+
+	ctx := context.Background()
+	c, err := cloudevents.NewClientHTTP(cloudevents.WithPort(env.Port))
+	if err != nil {
+		clog.Fatalf("failed to create event client, %v", err)
+	}
+	if err := c.StartReceiver(ctx, func(ctx context.Context, event cloudevents.Event) error {
+		// If this bot can handle the event, call the handler.
+
+		clog.FromContext(ctx).With("event", event).Debugf("received event")
+
+		defer func() {
+			if err := recover(); err != nil {
+				clog.Errorf("stacktrace from panic: %s", debug.Stack())
+			}
+		}()
+
+		if prb, ok := b.(interface {
+			OnPullRequest(ctx context.Context, pr *github.PullRequest) error
+		}); ok && event.Type() == "dev.chainguard.github.pull_request" {
+			log := clog.FromContext(ctx).With("bot", b.Name(), "event", event.Type())
```

Comment:
Is this the right place to start collecting metrics on bot response codes, latency, etc.?

Reply:
This is already collected for us by GCP! We can make dashboards for it, per-bot, or one dashboard for all bots.

Reply:
Ah, fantastic! I just couldn't find any dashboards that had anything besides the high level pubsub which is not that useful from the eventing side since you can't do things like, per event type, per bot, etc.

Reply:
If you can give me a pointer to something where they are, or what to look for, I could take a look at doing a dashboard tmw.